doc: add the upgrade guide from 2021.1 to 2022.1

annastuchlik opened the pull request:
Fix https://github.com/scylladb/scylla-docs/issues/4040
Fix https://github.com/scylladb/scylla-docs/issues/4128

This PR adds the upgrade guides from ScyllaDB Enterprise 2021.1 to 2022.1. They are based on the previous guides.

annastuchlik commented:
@tzach Could you review this PR? I've left you two questions above. Thanks.

tzach commented on docs/upgrade/_common/upgrade_to_2022_warning.rst:
```diff
@@ -0,0 +1,3 @@
+.. note:: Before upgrading to 2022.1, make sure to use `Scylla Monitoring 3.6 <https://github.com/scylladb/scylla-monitoring/releases/tag/scylla-monitoring-3.6>`_ or newer for the 2022.1 d.ashboards.
```
```suggestion
.. note:: Before upgrading to 2022.1, make sure to use `Scylla Monitoring 4.0 <https://github.com/scylladb/scylla-monitoring/releases/tag/scylla-monitoring-4.0.0>`_ or newer for the 2022.1 dashboards.
```

tzach commented on docs/upgrade/_common/upgrade-guide-from-2021.1-to-2022.1-ubuntu-and-debian.rst:
```diff
@@ -0,0 +1,206 @@
+
+Applicable Versions
+===================
+This guide covers upgrading ScyllaDB from version **2021.1.8** or later to ScyllaDB Enterprise version 2021.1.y on the following platform:
```
```suggestion
This guide covers upgrading ScyllaDB Enterprise from version 2021.1.x to ScyllaDB Enterprise version 2022.1.y on the following platform:
```

tzach commented on docs/upgrade/upgrade-enterprise/upgrade-guide-from-2021.1-to-2022.1/metric-update-2021.1-to-2022.1.rst:
```diff
@@ -0,0 +1,43 @@
+=========================================================================
+
+The following metrics are new in 2022.1 compared to 2021.1:
+
```
Add description per metric

annastuchlik commented on docs/upgrade/upgrade-enterprise/_common/enterprise_2022.1_warnings.rst:
```diff
@@ -0,0 +1,8 @@
+   If you are using CDC and upgrading ScyllaDB 2021.1 to 2022.1, please review the API updates in :doc:`querying CDC streams </using-scylla/cdc/cdc-querying-streams>` and :doc:`CDC stream generations </using-scylla/cdc/cdc-stream-generations>`.
+   In particular, you should update applications that use CDC according to :ref:`CDC Upgrade notes <scylla-4-3-to-4-4-upgrade>` **before** upgrading the cluster to 2022.1.
+
+.. include:: /upgrade/upgrade-enterprise/_common/mv_si_rebuild_warning.rst
```
From Tzach: remove it.

annastuchlik commented:
@tzach I've addressed your feedback: 
- Removed the warning about previous versions.
- Removed information for Ubuntu 16.04
- Added the description of new metrics

Could you review again?

tzach reviewed (approved):
LGTM

annastuchlik commented:
@nyh Could you merge this PR?

denesb commented:
There are some commits with `@users.noreply.github.com` email address:
```
commit 773be9fc025b53d28cf5afb383ef912c4e34ab33
Author: Anna Stuchlik <37244380+annastuchlik@users.noreply.github.com>
Date:   Thu Jul 14 10:24:27 2022 +0200

    Update docs/upgrade/_common/upgrade-guide-from-2021.1-to-2022.1-ubuntu-and-d
ebian.rst
    
    Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

commit bca51c9bb94f23bce6e16158b9f066647a91786f
Author: Anna Stuchlik <37244380+annastuchlik@users.noreply.github.com>
Date:   Thu Jul 14 09:17:51 2022 +0200

    Update docs/upgrade/_common/upgrade_to_2022_warning.rst
    
    Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>
```
Please reset the author of said commits so your own `@scylladb.com` email address appears as the author.

denesb commented:
In general avoid making edits via the in-browser Github UI, this will create commits with the above mentioned github.com email address.

nyh commented:
@denesb sorry, I commit this a few seconds before you noticed the addresses.

@benipeled FYI the pull script told me "WARNING: Some of the tests that ran for this PR were not completed successfully,", and yet there is a green checkmark saying "all checks have passed" (with 1 successful check). I know you're looking for examples of this, so here's another one.

tzach commented:
> In general avoid making edits via the in-browser Github UI, this will create commits with the above mentioned github.com email address.

Suggestions is a very useful feature. It would be a shame if we can not use it.

nyh commented:
> > In general avoid making edits via the in-browser Github UI, this will create commits with the above mentioned github.com email address.
> 
> Suggestions is a very useful feature. It would be a shame if we can not use it.

I think (but can be overruled by other maintainers, so let's hash it out here) that part of the decision to welcome the documentation into the main scylla.git was to turn a blind eye to *some* of the traditions that are different in that directory. For example, in code patches we don't like to have patch 1 write 20 lines of text and then patch 2 fix one of these lines of text - we like these patches to be squashed. But Anna and Tzach don't want to do that. And now they want to use a UI tool which results in weird github addresses. @tzach can you please check if this is an unavoidable consequence of the editing UI?

I asked Anna to please prefix all documentation commits with "doc: " precisely so that we (the maintainers) can care a bit less about the style of patches in that area. I wouldn't like to see a big mess, and if there's a way to improve the author listing situation let's do it (e.g., I asked Anna to fill her github name and she did and now my merge process is a bit less messy), but if it turns out that it's impossible and these ugly email addresses are something github cannot avoid, I vote not to forbid it.

Another option is to update our pull_github_pr.sh script to recognize these mis-addressed patches and fix them before pushing (i.e., make this into a rebase, not a merge). If @denesb or others think this is a big problem, we can definitely do that improvement to the script.

denesb commented:
I do not insist on enforcing the same rules on doc PRs, that we enforce on core patches. We already distinguish between patches based on their target. For example, patches targeting `scylla-gdb.py` or other utility scripts are usually less meticulously reviewed in terms of patches doing one thing, coding standards and so on. I'm fine with making a similar exception for docs, I was just not aware of this discussion that you mentioned (probably happened on a doc PR I didn't read through), and I defaulted on our base rules.

benipeled commented:
> @benipeled FYI the pull script told me "WARNING: Some of the tests that ran for this PR were not completed successfully,", and yet there is a green checkmark saying "all checks have passed" (with 1 successful check). I know you're looking for examples of this, so here's another one.

@nyh the warning message appears when the combined status [0] for a given commit (HEAD) is not _success_,
In this case, the script check [1] which has no statuses at all because docs use github-actions for verification that use **checks** and not **statuses** and as described on [0] the API return pending in case of no statuses,

I'm getting tired of all these tricks we're using on wrapper-scripts

[0] https://docs.github.com/en/rest/commits/statuses#get-the-combined-status-for-a-specific-reference
[1] https://api.github.com/repos/scylladb/scylla/commits/325437837574195e12dc821d7f5f053aa86903aa/status
